mcconf/skygauge_m_2814.h

```diff
@@ -0,0 +1,762 @@
+#ifndef MCCONF_FOC_ENCODER_INVERTED
+#define MCCONF_FOC_ENCODER_INVERTED 0
+#endif
+
+// Encoder Offset
+#ifndef MCCONF_FOC_ENCODER_OFFSET
+#define MCCONF_FOC_ENCODER_OFFSET 180
+#endif
+
+// Encoder Ratio
+#ifndef MCCONF_FOC_ENCODER_RATIO
+#define MCCONF_FOC_ENCODER_RATIO 7
+#endif
+
+// Sin/Cos Sine Gain Compensation
+#ifndef MCCONF_FOC_ENCODER_SIN_GAIN
+#define MCCONF_FOC_ENCODER_SIN_GAIN 1
+#endif
+
+// Sin/Cos Cosine Gain Compensation
+#ifndef MCCONF_FOC_ENCODER_COS_GAIN
+#define MCCONF_FOC_ENCODER_COS_GAIN 1
+#endif
+
+// Sin/Cos Sine Offset
+#ifndef MCCONF_FOC_ENCODER_SIN_OFFSET
+#define MCCONF_FOC_ENCODER_SIN_OFFSET 1.65
+#endif
+
+// Sin/Cos Cosine Offset
+#ifndef MCCONF_FOC_ENCODER_COS_OFFSET
+#define MCCONF_FOC_ENCODER_COS_OFFSET 1.65
+#endif
+
+// Sin/Cos Filter Constant
+#ifndef MCCONF_FOC_ENCODER_SINCOS_FILTER
+#define MCCONF_FOC_ENCODER_SINCOS_FILTER 0.5
+#endif
+
+// Sensor Mode
+#ifndef MCCONF_FOC_SENSOR_MODE
+#define MCCONF_FOC_SENSOR_MODE 0
+#endif
+
+// Speed Tracker Kp
+#ifndef MCCONF_FOC_PLL_KP
+#define MCCONF_FOC_PLL_KP 2000
+#endif
+
+// Speed Tracker Ki
+#ifndef MCCONF_FOC_PLL_KI
+#define MCCONF_FOC_PLL_KI 40000
+#endif
+
+// Motor Inductance (L)
+#ifndef MCCONF_FOC_MOTOR_L
+#define MCCONF_FOC_MOTOR_L 6.41e-06
+#endif
+
+// Motor Inductance Difference (Ld - Lq)
+#ifndef MCCONF_FOC_MOTOR_LD_LQ_DIFF
+#define MCCONF_FOC_MOTOR_LD_LQ_DIFF 0
+#endif
+
+// Motor Resistance (R)
+#ifndef MCCONF_FOC_MOTOR_R
+#define MCCONF_FOC_MOTOR_R 0.062
+#endif
+
+// Motor Flux Linkage (?)
+#ifndef MCCONF_FOC_MOTOR_FLUX_LINKAGE
+#define MCCONF_FOC_MOTOR_FLUX_LINKAGE 0.001459
+#endif
+
+// Observer Gain (x1M)
+#ifndef MCCONF_FOC_OBSERVER_GAIN
+#define MCCONF_FOC_OBSERVER_GAIN 4.9677e+08
+#endif
+
+// Observer Gain At Minimum Duty
+#ifndef MCCONF_FOC_OBSERVER_GAIN_SLOW
+#define MCCONF_FOC_OBSERVER_GAIN_SLOW 0.3
+#endif
+
+// Duty Downramp Kp
+#ifndef MCCONF_FOC_DUTY_DOWNRAMP_KP
+#define MCCONF_FOC_DUTY_DOWNRAMP_KP 10
+#endif
+
+// Duty Downramp Ki
+#ifndef MCCONF_FOC_DUTY_DOWNRAMP_KI
+#define MCCONF_FOC_DUTY_DOWNRAMP_KI 200
+#endif
+
+// Openloop ERPM
+#ifndef MCCONF_FOC_OPENLOOP_RPM
+#define MCCONF_FOC_OPENLOOP_RPM 1250
+#endif
+
+// Openloop ERPM at Min Current
+#ifndef MCCONF_FOC_OPENLOOP_RPM_LOW
+#define MCCONF_FOC_OPENLOOP_RPM_LOW 0.1
+#endif
+
+// D Axis Gain Scaling Start
+#ifndef MCCONF_FOC_D_GAIN_SCALE_START
+#define MCCONF_FOC_D_GAIN_SCALE_START 0.9
+#endif
+
+// D Axis Gain Scaling at Max Mod
+#ifndef MCCONF_FOC_D_GAIN_SCALE_MAX_MOD
+#define MCCONF_FOC_D_GAIN_SCALE_MAX_MOD 0.2
+#endif
+
+// Openloop Hysteresis
+#ifndef MCCONF_FOC_SL_OPENLOOP_HYST
+#define MCCONF_FOC_SL_OPENLOOP_HYST 0.1
+#endif
+
+// Openloop Lock Time
+#ifndef MCCONF_FOC_SL_OPENLOOP_T_LOCK
+#define MCCONF_FOC_SL_OPENLOOP_T_LOCK 0.5
+#endif
+
+// Openloop Ramp Time
+#ifndef MCCONF_FOC_SL_OPENLOOP_T_RAMP
+#define MCCONF_FOC_SL_OPENLOOP_T_RAMP 0.4
+#endif
+
+// Openloop Time
+#ifndef MCCONF_FOC_SL_OPENLOOP_TIME
+#define MCCONF_FOC_SL_OPENLOOP_TIME 0.1
```

Comment:
Add new define for soft-start current limit and soft-start target speed threshold